packages/contracts/src/managers/ZKEIP1271Manager.sol

```diff
@@ -21,7 +21,7 @@ abstract contract ZKEIP1271Manager is ZKAuth, ZKOwnerManager {
      * @param signature Signature of the data
      * @return magicValue Magic value if the signature is valid or invalid id / invalid time range
      */
-    function isValidSignature(bytes32 hash, bytes calldata signature) external view returns (bytes4 magicValue) {
+    function isValidSignature(bytes32 hash, bytes calldata signature) external pure returns (bytes4 magicValue) {
```

Comment:
This specification is defined in EIP1271 and this should be a view function.

Reply:
Thank you for the information.

> This specification is defined in EIP1271 and this should be a view function.

I referred to this: https://eips.ethereum.org/EIPS/eip-1271

I made these modifications following the Warning messages that were output when I compiled the code in my local environment.

```
> forge build
[⠘] Compiling...
[⠰] Compiling 2 files with 0.8.20
[⠔] Solc 0.8.20 finished in 28.71s
Compiler run successful with warnings:
Warning (2018): Function state mutability can be restricted to pure
  --> src/managers/ZKEIP1271Manager.sol:24:5:
   |
24 |     function isValidSignature(bytes32 hash, bytes calldata signature) external view returns (bytes4 magicValue) {
   |     ^ (Relevant source part starts here and spans across multiple lines).
```

Regarding the choice of visibility modifiers, I proposed the use of pure since the current code does not access any state variables. However, I will adhere to the proposal in EIP-1271.

Reply:
To clarify my understanding, is the following code intended for the todo section? This code is from the EIP-1271 proposal.

```
  /**
   * @notice Verifies that the signer is the owner of the signing contract.
   */
  function isValidSignature(
    bytes32 _hash,
    bytes calldata _signature
  ) external override view returns (bytes4) {
    // Validate signatures
    if (recoverSigner(_hash, _signature) == owner) {
      return 0x1626ba7e;
    } else {
      return 0xffffffff;
    }
  }
```